### PR TITLE
Lvnod 72 create eventDetails component

### PR DESCRIPTION
**Pull request by shyyka**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-XXX)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-XXX)

## Code reviewers

- [x] @yarvanya
- [x] @AlbaMontis
- [x] @avoytovich

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of issue

ToDo

## Summary of change

ToDo
## CHECK LIST
- [x]  СI passed
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR

### CODE REVIEW

File: app/src/containers/Event/EventDetails.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  componentWillMount() {
+    const data = this.props.events;
+    this.setState({ events: data.events });
+  }
+
+  render() {
+    const id = this.props.params.id;
+    const event = this.state.events.filter((item) => {
+      if (item.id == id) {
+        return item;
```
Comment (avoytovich) [inline app/src/containers/Event/EventDetails.js]: you will be right at all if do it throw findIndex...

File: app/src/redux/eventReducers.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const FETCH_EVENTS = 'FETCH_EVENTS';
+const FETCH_EVENTS_FULFILLED = 'FETCH_EVENTS_FULFILLED';
+const FETCH_EVENTS_REJECTED = 'FETCH_EVENTS_REJECTED';
+const apiUrl = 'http://localhost:8082';
```
Comment (yarvanya) [inline app/src/redux/eventReducers.js]: Delete this apiUrl, we will have this const in helper folder.

File: app/src/redux/eventReducers.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+import request from 'superagent';
+
+export const fetchEvents = () => {
+  return (dispatch) => {
+    request
+    .get(apiUrl + '/api/events')
```
Comment (yarvanya) [inline app/src/redux/eventReducers.js]: Use ".get(API.HOST + API.PORT + '/api/events')".

File: app/src/containers/Event/EventDetails.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  componentWillMount() {
+    const data = this.props.events;
+    this.setState({ events: data.events });
+  }
+
+  render() {
+    const id = this.props.params.id;
+    const event = this.state.events.filter((item) => {
```
Comment (AlbaMontis) [inline app/src/containers/Event/EventDetails.js]: You can omit () around item.

File: app/src/containers/Event/EventDetails.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            </div>
+            { this.props.children }
+          </Col>
+        </Row>
+      </Grid>
+    );
+  }
+}
+
+const mapStateToProps = (state, ownProps) => {
```
Comment (AlbaMontis) [inline app/src/containers/Event/EventDetails.js]: You can return object without "return", just `....=> ({prop: val})`

File: app/src/containers/Event/EventDetails.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  return {
+    events: state.events,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
```
Comment (AlbaMontis) [inline app/src/containers/Event/EventDetails.js]: The same as above. And useless ().

File: app/src/redux/eventReducers.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    case FETCH_EVENTS_REJECTED: {
+      return {
+              ...state,
+              fetching: false,
+              error: action.payload,
+            };
+    }
+  }
+  return state;
```
Comment (AlbaMontis) [inline app/src/redux/eventReducers.js]: Default

File: app/src/redux/eventReducers.js (hunk 3 above)
Comment (shyyka) [reply app/src/redux/eventReducers.js]: Done.

File: app/src/redux/eventReducers.js (hunk 2 above)
Comment (shyyka) [reply app/src/redux/eventReducers.js]: Done.

File: app/src/containers/Event/EventDetails.js (hunk 6 above)
Comment (shyyka) [reply app/src/containers/Event/EventDetails.js]: Done.

File: app/src/redux/eventReducers.js (hunk 7 above)
Comment (shyyka) [reply app/src/redux/eventReducers.js]: Done.

File: app/src/containers/Event/EventDetails.js (hunk 1 above)
Comment (shyyka) [reply app/src/containers/Event/EventDetails.js]: As for me it looks more simple and elegant)

**Review by yarvanya — DISMISSED** [review]

File: app/src/containers/Event/EventDetails.js (hunk 5 above)
Comment (shyyka) [reply app/src/containers/Event/EventDetails.js]: Done.

File: app/src/containers/Event/EventDetails.js (hunk 4 above)
Comment (shyyka) [reply app/src/containers/Event/EventDetails.js]: Done.

**Review by avoytovich — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

File: app/src/redux/reducer.js [hunk 8]
```diff
@@ -1,8 +1,10 @@
 import { combineReducers } from 'redux';
 import items from './Test';
+import eventReducers from './eventReducers';
 import { routerReducer } from 'react-router-redux';
 
 export default combineReducers({
   items,
-  routing: routerReducer
+  routing: routerReducer,
+  events: eventReducers,
```
Comment (4asof4ik) [inline app/src/redux/reducer.js]: check how include in other PR

File: app/src/redux/eventReducers.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+          type: FETCH_EVENTS_FULFILLED,
+          payload: res.body,
+        });
+      }
+    });
+  };
+};
+
+export const eventReducers = (
+  state = {
```
Comment (4asof4ik) [inline app/src/redux/eventReducers.js]: create let and include here

File: app/src/redux/eventReducers.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+              events: action.payload,
+            };
+    }
+
+    case FETCH_EVENTS_REJECTED: {
+      return {
+              ...state,
+              fetching: false,
+              error: action.payload,
+            };
```
Comment (4asof4ik) [inline app/src/redux/eventReducers.js]: check space in all PR

File: app/src/containers/Event/EventDetails.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as eventActions from '../../redux/eventReducers';
+
+class EventDetails extends React.Component {
+
+  constructor (props, context) {
+    super(props, context);
+    this.state = {
+      events: [],
```
Comment (4asof4ik) [inline app/src/containers/Event/EventDetails.js]: this is event detail but not events

check all file

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: app/src/redux/eventReducers.js (hunk 9 above)
Comment (shyyka) [reply app/src/redux/eventReducers.js]: Done.

File: app/src/redux/reducer.js (hunk 8 above)
Comment (shyyka) [reply app/src/redux/reducer.js]: Done.

File: app/src/redux/eventReducers.js (hunk 10 above)
Comment (shyyka) [reply app/src/redux/eventReducers.js]: Done.

File: app/src/containers/Gift/sampleGift.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { PageHeader, } from 'react-bootstrap';
```
Comment (AlbaMontis) [inline app/src/containers/Gift/sampleGift.js]: Useless ,

File: app/src/containers/Gift/sampleGift.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+import { Link } from 'react-router';
+
+export default class Gift extends React.Component {
+  render () {
+    return (
+      <div>
+        <PageHeader className="text-center">Here will be List of Gifts</PageHeader>
+          <div className="list-group">
```
Comment (AlbaMontis) [inline app/src/containers/Gift/sampleGift.js]: Fix indentation.

File: app/src/containers/Guest/sampleGuest.js [hunk 14]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { PageHeader, } from 'react-bootstrap';
```
Comment (AlbaMontis) [inline app/src/containers/Guest/sampleGuest.js]: Useless ,

File: app/src/containers/Guest/sampleGuest.js [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+import { Link } from 'react-router';
+
+export default class Guest extends React.Component {
+  render () {
+    return (
+      <div>
+        <PageHeader className="text-center">Here will be List of Guests</PageHeader>
+          <div className="list-group">
```
Comment (AlbaMontis) [inline app/src/containers/Guest/sampleGuest.js]: Fix indentation.

File: app/src/redux/eventReducers.js [hunk 16]
```diff
@@ -0,0 +1,62 @@
+const FETCH_EVENT_BY_ID = 'FETCH_EVENT_BY_ID';
+const FETCH_EVENT_BY_ID_FULFILLED = 'FETCH_EVENT_BY_ID_FULFILLED';
+const FETCH_EVENT_BY_ID_REJECTED = 'FETCH_EVENT_BY_ID_REJECTED';
+import  API  from '../helper/constants';
+
+import request from 'superagent';
+
+export const fetchEventById = eventId => {
+  return (dispatch) => {
```
Comment (AlbaMontis) [inline app/src/redux/eventReducers.js]: useless ().

File: app/src/redux/eventReducers.js [hunk 17]
```diff
@@ -0,0 +1,62 @@
+const FETCH_EVENT_BY_ID = 'FETCH_EVENT_BY_ID';
+const FETCH_EVENT_BY_ID_FULFILLED = 'FETCH_EVENT_BY_ID_FULFILLED';
+const FETCH_EVENT_BY_ID_REJECTED = 'FETCH_EVENT_BY_ID_REJECTED';
+import  API  from '../helper/constants';
+
+import request from 'superagent';
```
Comment (AlbaMontis) [inline app/src/redux/eventReducers.js]: Import from './../helper/request'

File: app/src/containers/Gift/sampleGift.js (hunk 13 above)
Comment (shyyka) [reply app/src/containers/Gift/sampleGift.js]: Done.

File: app/src/containers/Gift/sampleGift.js (hunk 12 above)
Comment (shyyka) [reply app/src/containers/Gift/sampleGift.js]: Done.

File: app/src/containers/Guest/sampleGuest.js (hunk 14 above)
Comment (shyyka) [reply app/src/containers/Guest/sampleGuest.js]: Done.

File: app/src/containers/Guest/sampleGuest.js (hunk 15 above)
Comment (shyyka) [reply app/src/containers/Guest/sampleGuest.js]: Done.

File: app/src/redux/eventReducers.js (hunk 16 above)
Comment (shyyka) [reply app/src/redux/eventReducers.js]: Done.

File: app/src/redux/eventReducers.js (hunk 17 above)
Comment (shyyka) [reply app/src/redux/eventReducers.js]: Done.

**Review by AlbaMontis — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

**Review by avoytovich — DISMISSED** [review]

File: app/src/containers/Event/EventDetails.js (hunk 11 above)
Comment (shyyka) [reply app/src/containers/Event/EventDetails.js]: Done.

File: app/src/containers/Event/EventDetails.js [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as eventActions from '../../redux/eventReducers';
+
+@connect( (state, ownProps) => ({ 
+  event: state.event
+  }), dispatch => ({  actions: bindActionCreators(eventActions, dispatch) }))
+
+export default class EventDetails extends React.Component {
+  constructor (props, context) {
```
Comment (AlbaMontis) [inline app/src/containers/Event/EventDetails.js]: It looks like you don't need constructor here.

File: app/src/redux/reducer.js [hunk 19]
```diff
@@ -2,9 +2,11 @@ import { combineReducers } from 'redux';
 import { routerReducer } from 'react-router-redux';
 import items from './Test';
 import login from './Login';
+import event from './eventReducers';
 
-export default combineReducers({
+export default combineReducers({
```
Comment (AlbaMontis) [inline app/src/redux/reducer.js]: Delete spaces at the end.

File: package.json [hunk 20]
```diff
@@ -20,6 +20,7 @@
   },
   "homepage": "https://github.com/Lv-237-NodeJS/da-mi#readme",
   "devDependencies": {
+    "babel-plugin-transform-decorators-legacy": "^1.3.4",
```
Comment (AlbaMontis) [inline package.json]: I think it should be in "dependencies".

Comment (shyyka) [reply package.json]: They recommend in dev - https://babeljs.io/docs/plugins/transform-decorat
I thik to dependenies

File: app/src/containers/Event/EventDetails.js (hunk 18 above)
Comment (shyyka) [reply app/src/containers/Event/EventDetails.js]: Yes.

File: app/src/redux/reducer.js (hunk 19 above)
Comment (shyyka) [reply app/src/redux/reducer.js]: Done.

**Review by AlbaMontis — DISMISSED** [review]

**Review by avoytovich — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

File: .babelrc [hunk 21]
```diff
@@ -1,3 +1,3 @@
 {
-  presets:["es2015", "stage-0", "react"]
+  presets:["es2015", "stage-0", "react"],
```
Comment (AlbaMontis) [inline .babelrc:2]: Useless ,

**Review by AlbaMontis — DISMISSED** [review]

File: .babelrc (hunk 21 above)
Comment (shyyka) [reply .babelrc:2]: Trailing commas makes version-control diffs cleaner, you can simply add a new line without modifying the previously last line ...

**Review by avoytovich — DISMISSED** [review]

File: app/src/containers/Event/EventDetails.js [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+            </div>
+            { this.props.children }
+          </Col>
+        </Row>
+      </Grid>
+    );
+  }
+}
+
+const mapStateToProps = (state, ownProps) => ({
```
Comment (4asof4ik) [inline app/src/containers/Event/EventDetails.js]: where you use ownProps

**Review by 4asof4ik — APPROVED** [review]

**Review by avoytovich — APPROVED** [review]

**Review by AlbaMontis — APPROVED** [review]

**Review by yarvanya — APPROVED** [review]

**Pull request merged**